src/extension.ts

```diff
@@ -1,173 +1,224 @@
+  );
+  const restartServer = vscode.commands.registerCommand(
+    "snooty.restart",
+    async () => {
+      await client.stop();
+      return client.start();
     }
-
-    const run: Executable = {
-        args: ['language-server'],
-        command: executableCommand
-    };
-    const debug: Executable = run;
-    const serverOptions: ServerOptions = {
-        run: run,
-        debug: debug
-    };
-
-    // client extensions configure their server
-    const documentSelector = [
-        { language: 'plaintext', scheme: 'file' },
-        { language: 'yaml', scheme: 'file' },
-        { language: 'restructuredtext', scheme: 'file' },
-        { language: 'toml', scheme: 'file' },
-    ];
-    const clientOptions: LanguageClientOptions = {
-        documentSelector,
-        synchronize: {
-            configurationSection: 'snooty',
-            fileEvents: [
-                vscode.workspace.createFileSystemWatcher('**/*.rst'),
-                vscode.workspace.createFileSystemWatcher('**/*.txt'),
-                vscode.workspace.createFileSystemWatcher('**/*.yaml'),
-                vscode.workspace.createFileSystemWatcher('snooty.toml')
-            ]
-        }
+  );
+
+  // Push the disposable to the context's subscriptions so that the
+  // client can be deactivated on extension deactivation
+  context.subscriptions.push(client.start());
+  context.subscriptions.push(restartServer);
+
+  // Register custom command to allow includes, literalincludes, and figures to be clickable
+  const clickInclude: vscode.Disposable = vscode.commands.registerCommand(
+    "snooty.clickInclude",
+    async (args) => {
+      const hoverFile = args.hoverFile;
+
+      // Send request to server (snooty-parser)
+      const type = mime.getType(hoverFile);
+
+      if (type == null || !type.includes("image")) {
+        const textDoc = await vscode.workspace.openTextDocument(hoverFile);
+        vscode.window.showTextDocument(textDoc);
+      } else {
+        open(hoverFile);
+      }
     }
+  );
+  context.subscriptions.push(clickInclude);
+
+  // Shows clickable link to file after hovering over it
+  vscode.languages.registerHoverProvider(
+    documentSelector,
+    new (class implements vscode.HoverProvider {
+      provideHover(
+        _document: vscode.TextDocument,
+        _position: vscode.Position,
+        _token: vscode.CancellationToken
+      ): vscode.ProviderResult<vscode.Hover> {
+        // Get range for a link
+        const wordRegex = /\/\S+/;
+        const wordRange = _document.getWordRangeAtPosition(
+          _position,
+          wordRegex
+        );
+
+        if (wordRange != undefined) {
+          // Get text at that range
+          let word = _document.getText(wordRange);
+
+          // If there's no extension, assume .txt.
+          // Ideally the parser could figure this out for us, but
+          // that's an issue for another time?
+          if (!word.includes(".")) {
+            word = word + ".txt";
```

Comment:
For context, I added this to support files under the `.. toctree::` directive (example [here](https://github.com/mongodb/docs-compass/blob/master/source/collections.txt#L152)) because these files are usually written without the extension and are always `.txt` files, and without adding this extension the parser would error out. Open to any suggestions, though, if you think this behavior shouldn't happen